apps/server/src/app/roles/providers/ability.factory.ts

```diff
@@ -0,0 +1,39 @@
+import { Ability, AbilityBuilder, AbilityClass, ExtractSubjectType } from "@casl/ability"
+import { Injectable } from "@nestjs/common"
+
+import { User, UserRoles } from "@app/users"
+import { Actions } from "../constants/actions";
+import { AppAbility, Subjects } from "../api";
+
+@Injectable()
+export class AbilityFactory {
+
+  defineAbility(user: User): AppAbility {
+
+    const { build, can, cannot } = new AbilityBuilder(Ability as AbilityClass<AppAbility>)
+
+    // should be role instead of userrole this is a bit duplicated
+    switch (user.role) {
+      case UserRoles.Admin:
+        can(Actions.Manage, 'all')
+        break
+
+      // can not update users
+      case UserRoles.Write:
+        can(Actions.Update, 'all')
```

Comment:
not all, Write has only rights to write into the
Set Parameter Entpoints in connection, but for example not 
for changing the connection params,...